src/main/java/mate/jdbc/service/CarServiceImpl.java

```diff
@@ -0,0 +1,71 @@
+package mate.jdbc.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import mate.jdbc.dao.CarDao;
+import mate.jdbc.lib.Inject;
+import mate.jdbc.lib.Service;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+
+@Service
+public class CarServiceImpl implements CarService {
+    @Inject
+    private CarDao carDao;
+
+    @Override
+    public Car create(final Car car) {
+        return carDao.create(car);
+    }
+
+    @Override
+    public Car get(final Long id) {
+        return carDao.get(id).orElseThrow(
+                () -> new RuntimeException("Could get Car by id: " + id + ". Value is null")
+        );
+    }
+
+    @Override
+    public List<Car> getAll() {
+        return carDao.getAll();
+    }
+
+    @Override
+    public Car update(final Car car) {
+        return carDao.update(car);
+    }
+
+    @Override
+    public boolean delete(final Long id) {
+        return carDao.delete(id);
+    }
+
+    @Override
+    public List<Car> getAllByDriver(final Long driverId) {
+        return carDao.getAllByDriver(driverId);
+    }
+
+    @Override
+    public void addDriverToCar(final Driver driver, final Car car) {
+        if (car.getDrivers() == null) {
+            List<Driver> drivers = new ArrayList<>();
+            drivers.add(driver);
+            car.setDrivers(drivers);
+            update(car);
+        } else {
+            final List<Driver> drivers = car.getDrivers();
+            if (!drivers.contains(driver)) {
+                drivers.add(driver);
+                update(car);
+            }
```

Comment:
First of all, there is no need for this check, and secondly, are you sure this code works correctly?

Reply:
Why this check is not needed, if the driver is already in the list for what need to perform any action?;

About how method works:
On Service side - The method changes the object.
On DAO Side - When I launch update method it updates data in "cars" then in "cars_drivers" table it marks old connections between car and driver in "is_deleted" column to "TRUE" and then creates new one;
It was said to like this in the videos. Do you have another advice how to do this?